jobs/build/build-sync/Jenkinsfile

```diff
@@ -0,0 +1,127 @@
+                        $class: 'hudson.model.ChoiceParameterDefinition',
+                        choices: commonlib.ocp4Versions.join('\n'),
+                        defaultValue: commonlib.ocp4DefaultVersion,
+                    ],
+                ]
+            ],
+            disableConcurrentBuilds(),
+            timestamps(),
+        ]
+    )
+
+    echo "Initializing ${BUILD_VERSION} sync: #${currentBuild.number}"
+
+    // doozer_working must be in WORKSPACE in order to have artifacts archived
+    MIRROR_WORKING = "${WORKSPACE}/MIRROR_working"
+    // Location of the SRC=DEST input file
+    OC_MIRROR_INPUT = "${MIRROR_WORKING}/oc_mirror_input"
+    // Location of the image stream to apply
+    OC_IS_OBJECT = "${MIRROR_WORKING}/release-is.yaml"
+    // Locally stored image stream stub
+    BASE_IMAGE_STREAM = "/home/jenkins/base-art-latest-imagestream.yaml"
+    // Kubeconfig allowing ART to interact with api.ci.openshift.org
+    CI_KUBECONFIG = "/home/jenkins/kubeconfigs/art-publish.kubeconfig"
+
+    // See 'oc image mirror --help' for more information.
+    // This is the template for the SRC=DEST strings mentioned above.
+    OC_FMT_STR = "registry.reg-aws.openshift.com:443/{repository}=quay.io/openshift-release-dev/ocp-v4.0-art-dev:{version}-{release}-{image_name_short}"
+
+    //Clear out previous work
+    sh "rm -rf ${MIRROR_WORKING}"
+    sh "mkdir -p ${MIRROR_WORKING}"
+
+    stage("Version dumps") {
+        buildlib.doozer """
+--version
+"""
+        sh "which doozer"
+        sh "oc version"
+    }
+
+    // TRY all of this so we can save the generated artifacts before
+    // throwing the exceptions
+    try {
+        // ######################################################################
+        // This should create a list of SOURCE=DEST strings in the output file
+        // May take a few minutes because doozer must query brew for each image
+        stage("Generate SRC=DEST input") {
+            buildlib.doozer """
+--working-dir "${MIRROR_WORKING}" --group 'openshift-${BUILD_VERSION}'
+beta:release-gen
+--sd ${OC_MIRROR_INPUT}
+--is ${OC_IS_OBJECT}
+--isb ${BASE_IMAGE_STREAM}
+'${OC_FMT_STR}'
+"""
+            images_count = sh(returnStdout: true, script: "wc -l ${OC_MIRROR_INPUT}").trim().split().first()
+            currentBuild.description = "Preparing to sync ${images_count} images"
+        }
+
+        // ######################################################################
+        // Now run the actual mirroring command. Wrapped this in a
+        // retry loop because it is known to fail occassionally
+        // depending on the health of the source/destination endpoints.
+        stage("oc image mirror") {
+            echo "Mirror SRC=DEST input:"
+            sh "cat ${OC_MIRROR_INPUT}"
+            try {
+                retry (3) {
+                    buildlib.registry_quay_dev_login()
+                    buildlib.oc """
+image
```

Comment:
Could this not be done with doozer now that it uses the same command?

Reply:
Good question. For **right now** the answer is no. I will comment in more detail later explaining why.

Reply:
Figured, but wanted to check.